// coding=utf-8
// Copyright 2022 Foivos Tsimpourlas and Chris Cummins.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto2";

package benchpress;

option go_package = "benchpresspb";
option java_multiple_files = true;
option java_outer_classname = "TelemetryProto";
option java_package = "com.benchpress";

// Records telemetry data about a single epoch of model training.
message ModelEpochTelemetry {
  // The number of milliseconds since the epoch that epoch training completed.
  optional string timestamp_unix_epoch_ms = 1;
  // The epoch which has just finished training, starting at one.
  optional int32 epoch_num = 2;
  // The wall time that it took to train the epoch.
  optional int64 epoch_wall_time_ms = 3;
  // The model's loss.
  optional double loss = 4;
}
